tracing/tracing.go

```diff
@@ -6,51 +6,185 @@ package tracing
 
 import (
 	"context"
-	"io"
+	"fmt"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
 
 	"github.com/opentracing/opentracing-go"
 	"github.com/pkg/errors"
-	jaeger "github.com/uber/jaeger-client-go"
-	jaegercfg "github.com/uber/jaeger-client-go/config"
-	jaegerprom "github.com/uber/jaeger-lib/metrics/prometheus"
+	"github.com/uber/jaeger-client-go"
+	jaegerpropagator "go.opentelemetry.io/contrib/propagators/jaeger"
+	"go.opentelemetry.io/contrib/samplers/jaegerremote"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	bridge "go.opentelemetry.io/otel/bridge/opentracing"
+	jaegerotel "go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/propagation"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
 )
 
 // ErrInvalidConfiguration is an error to notify client to provide valid trace report agent or config server
 var (
 	ErrBlankTraceConfiguration = errors.New("no trace report agent, config server, or collector endpoint specified")
 )
 
-// installJaeger registers Jaeger as the OpenTracing implementation.
-func installJaeger(serviceName string, cfg *jaegercfg.Configuration, options ...jaegercfg.Option) (io.Closer, error) {
-	metricsFactory := jaegerprom.New()
+const (
+	envSamplerParam           = "JAEGER_SAMPLER_PARAM"
+	envEndpoint               = "JAEGER_ENDPOINT"
+	envJaegerAgentHost        = "JAEGER_AGENT_HOST"
+	envJaegerAgentPort        = "JAEGER_AGENT_PORT"
+	envSamplerType            = "JAEGER_SAMPLER_TYPE"
+	envSamplerRemoteURL       = "JAEGER_SAMPLING_ENDPOINT"
+	DefaultSamplingServerPort = 5778
+)
+
+func NewFromEnv(serviceName string) (*tracesdk.TracerProvider, error) {
+	cfg, err := parseTracingConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not load jaeger tracer configuration")
+	}
+	if cfg.samplerRemoteURL == "" && cfg.host == "" && cfg.collectorURL == "" {
+		return nil, ErrBlankTraceConfiguration
+	}
+	return cfg.initJaegerTracerProvider(serviceName)
+}
 
-	// put the metricsFactory earlier so provided options can override it
-	opts := append([]jaegercfg.Option{jaegercfg.Metrics(metricsFactory)}, options...)
+// Parse Jeager tags from env var JAEGER_TAGS, example of TAGs format: key1=value1,key2=value2
```

Comment:
s/Jeager/Jaeger/
Probably worth mentioning that `${FOO}` sequences are expanded as environment variables.